infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InLogEntrySyncState.java

```diff
@@ -148,8 +152,15 @@ public void onEntry(LogReplicationState from) {
                 fsm.getAckReader().markSnapshotSyncInfoCompleted();
             }
 
-            logEntrySyncFuture = fsm.getLogReplicationFSMWorkers().submit(() ->
-                    logEntrySender.send(transitionEventId));
+            if (from.getType() == LogReplicationStateType.IN_LOG_ENTRY_SYNC && !logEntrySender.isTaskActive()) {
+                logEntrySyncFuture = fsm.getLogReplicationFSMWorkers().schedule(() ->
+                        logEntrySender.send(transitionEventId), waitRetryRead, TimeUnit.SECONDS);
+                waitRetryRead = Math.min(waitRetryRead + 1, WAIT_RETRY_READ_MAX);
+            } else {
+                logEntrySyncFuture = fsm.getLogReplicationFSMWorkers().submit(() ->
+                        logEntrySender.send(transitionEventId));
+                waitRetryRead = 1;
```

Comment:
this must also be reset when a new cycle of log entry sync starts.  When from.getType() != to.getType()

Reply:
+1